### PR TITLE
Ckb auth cli

### PR DESCRIPTION
**Pull request by contrun**
See the ckb-auth-cli.md for usage.

### CODE REVIEW

File: tests/auth_rust/src/bin/ckb-auth-cli.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+fn parse_address(blockchain: &str, address: &str) {
+    println!(
+        "{}",
+        hex::encode(get_pub_key_hash_from_address(blockchain, address).expect("get pub key hash"))
+    );
+}
+
+fn generate_message(_blockchain: &str, pubkeyhash: Vec<u8>) {
+    let algorithm_type = AlgorithmType::Bitcoin;
```
Comment (15168316096) [inline tests/auth_rust/src/bin/ckb-auth-cli.rs:131]: "Why is the enumeration type not 'litecoin' even though it was forked from 'bitcoin'?"

Comment (contrun) [reply tests/auth_rust/src/bin/ckb-auth-cli.rs]: Thanks for spotting that error. As for use `enum`, this is useful, but not urgent.

File: docs/ckb-auth-cli.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+We can now sign the message with native blockchain clients. Depending on the specific blockchains,
+each blockchain wallets accept different parameters and output different data. See below for each
+blockchains.
+
+## Verify the signature with `verify` subcommand
+ckb-auth-cli generate the same transaction and set the signature to the one generated above,
+and then it checks the validity of this transaction.
+
+# integrations
+##  litecoin
```
Comment (XuJiandong) [inline docs/ckb-auth-cli.md:25]: Move these litecoin related to litecoin.md.

File: docs/ckb-auth-cli.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# ckb-auth-cli
```
Comment (XuJiandong) [inline docs/ckb-auth-cli.md:1]: Should include generic steps: it's not litecoin only.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: docs/ckb-auth-cli.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ckb-auth-cli is a command line utillity to faciliate the creation and verification of transaction
+with ckb-auth as lock script.
+
+## Get the pub key hash with `generate` sub command.
```
Comment (XuJiandong) [inline docs/ckb-auth-cli.md:5]: not 'generate', it's 'parse'.

File: docs/ckb-auth-cli.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+## Verify the signature with `verify` subcommand
+ckb-auth-cli generate the same transaction and set the signature to the one generated above,
+and then it checks the validity of this transaction.
+
+# integrations
+##  litecoin
+Below uses the sample key in [litecoin docs](./litecoin.md))
+
+### Get the pub key hash with `generate` sub command.
```
Comment (XuJiandong) [inline docs/ckb-auth-cli.md:28]: 'parse'

**Review by XuJiandong — CHANGES_REQUESTED** [review]

**Review by XuJiandong — APPROVED** [review]

**Pull request merged**